### PR TITLE
utils: Add `getBallotStyleGroup` helper

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

For cases where we have a group ID and want to look it up. I realized I needed this when I was rebasing another PR after the recent refactor to ballot styles.
## Demo Video or Screenshot
N/A

## Testing Plan
Added new tests

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/utils/src/ballot_styles.ts [hunk 1]
```diff
@@ -36,6 +37,7 @@ export function generateBallotStyleId(params: {
   languages: LanguageCode[];
   party?: Party;
 }): BallotStyleId {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```
Comment (carolinemodic) [inline libs/utils/src/ballot_styles.ts:40]: did you need to add this for it to clear lint? You didn't change the next line so checking. I was having issues with this lint rule on my PR (I dont remember if it was here specifically)

File: libs/utils/src/tabulation/lookups.ts [hunk 2]
```diff
@@ -149,12 +149,10 @@ export function determinePartyId<T>(
...
-    electionDefinition.election.ballotStyles
-  ).find((bs) => bs.id === group.ballotStyleGroupId);
-  if (!ballotStyle) return undefined;
-
-  return ballotStyle.partyId;
+  const ballotStyleGroup = getBallotStyleGroup({
+    election: electionDefinition.election,
+    ballotStyleGroupId: group.ballotStyleGroupId as BallotStyleGroupId,
+  });
+  return ballotStyleGroup?.partyId;
```
Comment (carolinemodic) [inline libs/utils/src/tabulation/lookups.ts:157]: I thought it was against the lint rules to use the non-null ? assertion. Did I just totally make that up?

**Review by carolinemodic — APPROVED** [review]

File: libs/utils/src/ballot_styles.ts (hunk 1 above)
Comment (jonahkagan) [reply libs/utils/src/ballot_styles.ts:40]: Yeah I couldn't pass lint without this comment

File: libs/utils/src/tabulation/lookups.ts (hunk 2 above)
Comment (jonahkagan) [reply libs/utils/src/tabulation/lookups.ts:157]: We can't use `!` but we can use `?`

**Pull request merged**